src/main/resources/assets/src/managerView/branchHoursView.js

```diff
@@ -1,13 +1,283 @@
+                })
+            });
+        } else {
+            await $.ajax({
+                type: 'POST',
+                headers: {
+                    'Accept': 'application/json',
+                    'Content-Type': 'application/json'
+                },
+                url: '/api/branchHours/update',
+                data: JSON.stringify({
+                    openTime: openTime,
+                    closeTime: closeTime,
+                    branchId: branchId,
+                    dayOfWeek: dayOfWeek,
+                    id: id,
+                })
+            });
+        }
+
+        await this.props.loadData('branchHours');
+        this.props.setStateValue('loading', false);
+
+        this.setState({
+            showEditDialog: false,
+        })
+    }
+
+    closeEditor() {
+        this.setState({
+            showEditDialog: false,
+        })
+    }
+
+    setFilters(newFilters) {
+        this.setState({
+            filters: newFilters,
+        })
+    }
+
+    clearFilters() {
+        this.setState({
+            filters: [[], [], [], []],
+        })
+    }
+
+    getBranchName(id) {
+        return this.props.branches.find(b => { return b.id == id }).name;
+    }
+
+    convertIntToDay(dayNum) {
+        switch(dayNum){
+            case 1:
+                return 'Sunday';
+            case 2:
+                return 'Monday';
+            case 3:
+                return 'Tuesday';
+            case 4:
+                return 'Wednesday';
+            case 5:
+                return 'Thursday';
+            case 6:
+                return 'Friday';
+            case 7:
+                return 'Saturday';
+        }
+    }
+
+    convertDayToInt(dayName) {
+        switch(dayName){
+            case 'Sunday':
+                return 1;
+            case 'Monday':
+                return 2;
+            case 'Tuesday':
+                return 3;
+            case 'Wednesday':
+                return 4;
+            case 'Thursday':
+                return 5;
+            case 'Friday':
+                return 6;
+            case 'Saturday':
+                return 7;
+        }
+    }
+
+    getDate(id) {
+        const date = this.props.calendar.find(c => { return c.calendarId == id }).date;
+        return date[1] + "-" + date[2] + "-" + String(date[0]).slice(2);
     }
 
     render() {
+        const headerNames = ['Open Time', 'Close Time', 'Branch', 'Day Of Week'];
+
+        // Filter options
+        const filterOptions = [
+            getTopResults(this.props.branchHours.map(a => a.openTime[0])).map(r => convertTime24to12(r.item)),
+            getTopResults(this.props.branchHours.map(a => a.closeTime[0])).map(r => convertTime24to12(r.item)),
+            getTopResults(this.props.branchHours.map(b => b.branchId)).map(r => this.getBranchName(r.item)),
+            getTopResults(this.props.branchHours.map(h => h.dayOfWeek)).map(r => this.convertIntToDay(r.item))];
```

Comment:
I beleive that `r.item` here is a string. In `convertIntToDay` it is checking against integers. So either convert it to integers or check for strings in the function. You can use `parseInt()`.

Reply:
Okay, the editing works fine now, but when I go to filter by Day of Week, it doesn't do anything. Is this done through the filterOptions const and am I inputting the wrong type of data?

Reply:
The actually filtering is done on lines 243-248.

Reply:
Ok. Not sure what is going wrong, the filtering is working for everything except day of week.

Reply:
Because you are doing table length - 1 in the for loop